### PR TITLE
merge latest wasmtime, fixup wasi integration and tests

### PR DESCRIPTION
**Pull request by pchickey**
* the `lucet-wasi` test suite now uses `cap-tempfile` most places! we needed a `Dir` so why not use the better api anyway?
* various errnos have changed in the test suite.
* lucet-wasi's getrusage test no longer makes sense - wasi doesnt pretend to support the process/thread timers anymore. So, I have deleted it.
* I renovated lucet-wasi's `fs.c` test to give unique failures depending where it fails, and isolated the particular behavior I was working on there into the `readdir.c` test
* raw-cpuid has been upgraded to 9.0.0 to resolve a RUSTSEC advisory. we also needed to bump the wasmtime submodule to get the fixed cranelift-native, which removes its dep on a vulnerable raw-cpuid

### CODE REVIEW

File: lucet-wasi/src/c_api.rs [hunk 1]
```diff
@@ -27,15 +27,18 @@ pub unsafe extern "C" fn lucet_wasi_ctx_args(
     assert_nonnull!(wasi_ctx);
     let mut b = Box::from_raw(wasi_ctx as *mut WasiCtxBuilder);
     let args_raw = std::slice::from_raw_parts(argv, argc);
-    let args: Result<Vec<&str>, _> = args_raw
+    let args: Result<Vec<String>, std::str::Utf8Error> = args_raw
         .iter()
-        .map(|arg| CStr::from_ptr(*arg).to_str())
+        .map(|arg| Ok(CStr::from_ptr(*arg).to_str()?.to_owned()))
```
Comment (acfoltzer) [inline lucet-wasi/src/c_api.rs:32]: This seems fine, I'm just curious what motivated the switch?

File: lucet-wasi/tests/guests/fs.c [hunk 2]
```diff
@@ -140,19 +224,31 @@ int main(void)
...
+    }
     res = mkdir("/sandbox/a/b/c", 0755);
-    assert(res == 0);
+    if (res != 0) {
+        exit(46);
+    }
     res = access("/sandbox/a/b/c", R_OK);
-    assert(res == 0);
+    if (res != 0) {
+        exit(47);
```
Comment (acfoltzer) [inline lucet-wasi/tests/guests/fs.c:250]: lol I hate this but I love it

File: lucet-wasi/tests/guests/readdir.c [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    res = 0;
+    struct dirent * entry = NULL;
+    while ((entry = readdir(dir)) != NULL) {
+        if (entry->d_name[0] == '.') {
+            res += 1000;
+        } else {
+            res++;
+        }
```
Comment (acfoltzer) [inline lucet-wasi/tests/guests/readdir.c:38]: oh my god

**Review by acfoltzer — APPROVED** [review]
Looks great, thank you!

File: lucet-wasi/tests/guests/readdir.c (hunk 3 above)
Comment (pchickey) [reply lucet-wasi/tests/guests/readdir.c:38]: lol

File: lucet-wasi/src/c_api.rs (hunk 1 above)
Comment (pchickey) [reply lucet-wasi/src/c_api.rs:32]: Honestly, I just forgot to make the args and env builders take an `AsRef<str>` and didn't want to make another upstream PR just to get this in. There wasn't any good reason for it.

File: lucet-wasi/tests/guests/fs.c (hunk 2 above)
Comment (pchickey) [reply lucet-wasi/tests/guests/fs.c:250]: once you get used to wasmtime having backtraces its real painful to go back

**Pull request merged**